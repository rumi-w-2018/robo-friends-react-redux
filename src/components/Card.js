import React from 'react';

const baseUrl = 'https://robohash.org/';

const Card = ({ name, email }) => {

    return(
        <div className='tc bg-washed-green br3 pa3 ma2 bw2 dib b--black-102 grow'>
            <img src={`${baseUrl}${name}?size=200x200`} alt='robot' />
            <div className='pa2 bt b--black-20'>
                <span className='tc pv1 db bg-animate bg-dark-blue hover-bg-blue white f5 br1' href='#'>                {name}
                </span>
                <p className='f6 black mv1'>{email}</p>
            </div>
        </div>
    );
};

export default Card;