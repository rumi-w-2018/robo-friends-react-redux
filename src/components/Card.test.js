import { shallow, mount, render } from 'enzyme';
import React from 'react';
import Card from './Card';

// shallow renderer renders only Card component
it('expects to render Card component', () => {
    expect(shallow(<Card />).length).toEqual(1);

});

// this create snapshot folder
it('expects to render Card component', () => {
    expect(shallow(<Card />)).toMatchSnapshot();
});