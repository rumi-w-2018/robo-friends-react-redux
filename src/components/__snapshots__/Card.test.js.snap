// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expects to render Card component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt="robot"
          src="https://robohash.org/undefined?size=200x200"
/>,
        <div
          className="pa2 bt b--black-20"
>
          <span
                    className="tc pv1 db bg-animate bg-dark-blue hover-bg-blue white f5 br1"
                    href="#"
          >
                                    
          </span>
          <p
                    className="f6 black mv1"
          />
</div>,
      ],
      "className": "tc bg-washed-green br3 pa3 ma2 bw2 dib b--black-102 grow",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "robot",
          "src": "https://robohash.org/undefined?size=200x200",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span
              className="tc pv1 db bg-animate bg-dark-blue hover-bg-blue white f5 br1"
              href="#"
>
                              
</span>,
            <p
              className="f6 black mv1"
/>,
          ],
          "className": "pa2 bt b--black-20",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "                ",
                undefined,
              ],
              "className": "tc pv1 db bg-animate bg-dark-blue hover-bg-blue white f5 br1",
              "href": "#",
            },
            "ref": null,
            "rendered": Array [
              "                ",
              undefined,
            ],
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "f6 black mv1",
            },
            "ref": null,
            "rendered": null,
            "type": "p",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="robot"
            src="https://robohash.org/undefined?size=200x200"
/>,
          <div
            className="pa2 bt b--black-20"
>
            <span
                        className="tc pv1 db bg-animate bg-dark-blue hover-bg-blue white f5 br1"
                        href="#"
            >
                                        
            </span>
            <p
                        className="f6 black mv1"
            />
</div>,
        ],
        "className": "tc bg-washed-green br3 pa3 ma2 bw2 dib b--black-102 grow",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "robot",
            "src": "https://robohash.org/undefined?size=200x200",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                className="tc pv1 db bg-animate bg-dark-blue hover-bg-blue white f5 br1"
                href="#"
>
                                
</span>,
              <p
                className="f6 black mv1"
/>,
            ],
            "className": "pa2 bt b--black-20",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "                ",
                  undefined,
                ],
                "className": "tc pv1 db bg-animate bg-dark-blue hover-bg-blue white f5 br1",
                "href": "#",
              },
              "ref": null,
              "rendered": Array [
                "                ",
                undefined,
              ],
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "f6 black mv1",
              },
              "ref": null,
              "rendered": null,
              "type": "p",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
